### DISCUSSION TITLE
Should we let go of recipe ratings? 🤔

### DISCUSSION THREAD

**Discussion by Vianpyro** [TheSmartCooking/.github#1] [Ideas]
### Hi team! 👋  

As we continue refining features for Smart Cooking, I've been reflecting on whether we truly need the **recipe ratings** feature. Currently, we already track **likes** and **unique views** for each recipe. These two metrics might already give users a good sense of a recipe's popularity and quality.  

Instead of maintaining a separate rating system, here’s an alternative:  
- **Popularity Insight**: We could calculate a **likes/views ratio** as a lightweight way to infer recipe quality. For example, if a recipe has 50 likes and 200 views, the ratio (25%) could indicate user approval at a glance.  
- **Simplicity**: By removing the ratings feature, we could streamline both the user interface and backend logic, reducing complexity for users and developers alike.  

---

## 🛠️ Impact on the Database:  
If we decide to remove ratings, we’d delete this table entirely:  
```sql
CREATE OR REPLACE TABLE recipe_rating (
    rating_id INT AUTO_INCREMENT PRIMARY KEY,
    person_id INT,
    recipe_id INT,
    rating TINYINT CHECK (rating BETWEEN 1 AND 4),
    FOREIGN KEY (person_id) REFERENCES person (person_id) ON DELETE CASCADE,
    FOREIGN KEY (recipe_id) REFERENCES recipe (recipe_id) ON DELETE CASCADE,
    UNIQUE (person_id, recipe_id)
) ENGINE = InnoDB;
```

---

## 🤔 Questions to Consider:
1. Do we think likes and views are enough to gauge a recipe’s quality?  
2. Should we calculate and display a **likes/views ratio** as an additional metric?  
3. Do we lose anything meaningful by dropping ratings entirely?  
4. Would simplifying this feature improve user experience and development efficiency?  

---

## 🗳️ Let’s Discuss!  
Please share your thoughts! Are there any scenarios where recipe ratings add significant value? Or is it time to simplify and move forward with likes and views alone?  

Looking forward to hearing everyone’s input! 😊

**Comment by MBlueEyed** [comment]
1. Likes and views are a good start 
2. For admins yes, for users no (just number of likes and saved)
3. No, however I would recommend having ratings for users to rate the difficulty of the recipes 
4. Yes, but without ratio feature as well.
